### PR TITLE
Please review my changes. Fixed some typos and add ability to run with access point mode

### PR DESCRIPTION
**Pull request by pastukhov**

### CODE REVIEW

**Comment by marcoskirsch** [comment]
Hey some of the files do not diff properly, I am not sure why, maybe because the editor changed the kind of line endings being used. Do you mind looking into it?

**Comment by pastukhov** [comment]
I fixed ends of lines. Please review.

File: init.lua [hunk 1]
```diff
@@ -17,7 +25,7 @@ local compileAndRemoveIfNeeded = function(f)
    end
 end
 
-local serverFiles = {'httpserver.lua', 'httpserver-request.lua', 'httpserver-static.lua', 'httpserver-error.lua'}
+local serverFiles = {'httpserver.lua', 'httpserver-request.lua', 'httpserver-static.lua', 'httpserver-error.lua','http/ual.lua'}
```
Comment (marcoskirsch) [inline init.lua:28]: Things inside http/ aren't considered "server files", shouldn't be in this list.

File: init.lua [hunk 2]
```diff
@@ -1,10 +1,18 @@
...
+--wifi.setmode(wifi.STATIONAP)
 wifi.setmode(wifi.STATION)
-print('set mode=STATION (mode='..wifi.getmode()..')')
+print('set (mode='..wifi.getmode()..')')
 print('MAC: ',wifi.sta.getmac())
 print('chip: ',node.chipid())
 print('heap: ',node.heap())
-wifi.sta.config("Internet","")
+
+local joincounter = 0
```
Comment (marcoskirsch) [inline init.lua:9]: please declare this close to where it's being used, in line 37.

File: init.lua [hunk 3]
```diff
@@ -26,13 +34,18 @@ serverFiles = nil
...
+      joincounter = joincounter +1
+   else 
       tmr.stop(0)
-      print('IP: ',wifi.sta.getip())
+      -- print('IP: ',wifi.sta.getip())
       -- Uncomment to automatically start the server in port 80
-      -- dofile("httpserver.lc")(80)
+      joincounter = nil
+      collectgarbage()
+      dofile("httpserver.lc")(80)
```
Comment (marcoskirsch) [inline init.lua:47]: I'd like to leave this commented out in the repository, to match the comment but also because it's also easier for development.

**Comment by marcoskirsch** [comment]
Please bear with me as I learn to use git and github. I can lift some of the more obvious changes (those typos, what was I thinking!) right away on my own or I can try and figure out how to do the manual merge. I'll look into it.

**Comment by marcoskirsch** [comment]
Alright, I have made some of the small changes on my side. I committed each group of changes as it's own commit. It may be easiest for you to pull the latest master and re-do the pull request.

**Comment by pastukhov** [comment]
- Marcos, i use latest nodemcu-uploader.py from https://github.com/kmpm/nodemcu-uploader, latest commit 5c4dd0ed3638dd678ddaae7ade6a16563423254c and assume that it in the PATH
- upadeaplist.lua should return an json with list of access points available or store it at http/aplist.json, but at this moment it just reboot the mcu. it should be called with http://ipofnodemcu/updateaplist.lc
- same for me with git and github. I will close the pull request.

**Pull request closed without merging**